src/pudl/scripts/dbt_helper.py

```diff
@@ -167,6 +167,131 @@ def from_yaml(cls, schema_path: Path) -> "DbtSchema":
             return cls.model_validate(yaml.safe_load(schema_yaml))
 
 
+def diff_scalar(field, old, new):
+    """Return a diff of a scalar field value as a nested dictionary."""
+    return {field: {"old": old, "new": new}} if old != new else {}
+
+
+def diff_list(field, old, new):
+    """Return a diff of a list field value as a nested dictionary."""
+    old_set, new_set = set(old or []), set(new or [])
+    added, removed = list(new_set - old_set), list(old_set - new_set)
+    return {field: {"added": added, "removed": removed}} if added or removed else {}
+
+
+def diff_dict_keys(field, old, new):
+    """Return a diff of dictionary keys as a nested dictionary."""
+    old_keys, new_keys = set((old or {}).keys()), set((new or {}).keys())
+    added, removed = list(new_keys - old_keys), list(old_keys - new_keys)
+    return {field: {"added": added, "removed": removed}} if added or removed else {}
+
+
+def diff_dbt_column(o, n):
```

Comment:
Let's name these `old` and `new`, I was confused by the 1-letter names.

It would be nice, in general, to have some type hints here - I find them very helpful when thinking about what a function does.